applies enhancements

kzidane opened the pull request:
apologies for the broad commit/PR! after more testing, could detect many problems. went back to the docs and read through several more times only to realize that there were many flaws in my understanding of `Editor`, `Document`, `Session`, etc, what they were, and how they related and interacted.

had to rewrite almost the whole plugin. it should be much less buggy now! in summary, this commit:
- removes HTML and creates elements using JS.
- removes unneeded dependencies.
- tidies and inserts more comments.
- allows for simultaneous playing of audio files.
- starts audio players paused.
- fixes bugs and preserves state when moving tabs to different panes.
- ensures path is watched once and unwatched when it should.
- handles deletion or external renaming of audio tabs only.
- cleans up on document unload.

Commit:
applies enhancements

* removes HTML and creates elements using JS.
* removes unneeded dependencies.
* tidies and inserts more comments.
* allows for simultaneous playing of audio files.
* starts audio players paused.
* fixes bugs and preserves state when moving tabs to different panes.
* ensures path is watched once and unwatched when it should.
* handles deletion or external renaming of audio tabs only.
* cleans up on document unload.

kzidane commented on audioplayer.js:
```diff
@@ -1,135 +1,208 @@
-                setPath(path, audioDoc);
+                // alert user if file was deleted before unwatching
+                watcher.check(path);
+                setTimeout(function() {
```
@dmalan @danallan when file gets deleted while open, it gets unloaded right away. and since we wanna unwatch paths on unload, path gets unwatched before [`delete`](https://apidoc.c9.io/c9v3/#!/api/watcher-event-delete) is emitted by [`watcher`](https://apidoc.c9.io/c9v3/#!/api/watcher) causing tab to close immediately without alerting the user (per [#145](https://github.com/kzidane/harvard.cs50.audioplayer/blob/10f713e0df5e9ddaccd7747924f1b6e91197b996/audioplayer.js#L145)).

had to [`check`](https://apidoc.c9.io/c9v3/#!/api/watcher-method-check) to get watcher to emit `delete` and unwatch after timeout to ensure that the user was alerted before tab is closed.

not comfortable with it, but not sure if there's a better way. thoughts?

danallan replied:
Hmmmm. Do we want to override the default behavior here? Maybe I'm misreading the code, but if the user deletes the file in the GUI they will get two dialogs: a file deletion confirmation and then a file went away dialog, the latter from this plugin? I would think only the first dialog is necessary since that's clear enough about what's happened. If we want to override the behavior we should probably do it at a lower level so that it applies to all editors?

kzidane replied:
good point! removed per [3a212b4](https://github.com/cs50/harvard.cs50.audioplayer/pull/1/commits/3a212b4a8a0690a4d7c68a1f5071a50617d470f3)!

kzidane commented on audioplayer.js:
```diff
@@ -1,135 +1,208 @@
+                var tab = audioDoc.tab;
+                var path = tab.path;
+                var session = audioDoc.getSession();
+
```
``` javascript
if (!_.isObject(audioDoc) || !_.isObject(audioDoc.tab) || !_.isString(audioDoc.tab.path)) {
    return;
}
```

longer but better?

danallan replied:
Yep probably best to check early for requirements.

Commit:
checks requirements & removes redundant calls

in setPath, before trying to access Document's tab, and tab's path,
this ensures that these properties exist.

it also removes calls to check and setTimeout when handling
documentUnload, since no need to alert user about file unavailability
in this case as they confirm deletion by default.
